includes/widgets/common.php

```diff
@@ -981,6 +981,498 @@ protected function register_controls() {
+					],
+					'frontend_available' => true,
+				]
+			);
+
+			$this->end_popover();
+
+			$this->add_control(
+				"_{$tab}_transform_skew_popover",
+				[
+					'label' => __( 'Skew', 'elementor' ),
+					'type' => Controls_Manager::POPOVER_TOGGLE,
+					'prefix_class' => 'e-',
+					'return_value' => 'transform',
+				]
+			);
+
+			$this->start_popover();
+
+			$this->add_responsive_control(
+				"_{$tab}_transform_skewX_effect",
+				[
+					'label' => __( 'Skew X', 'elementor' ),
+					'type' => Controls_Manager::SLIDER,
+					'range' => [
+						'px' => [
+							'min' => -360,
+							'max' => 360,
+						],
+					],
+					'default' => [
+						'size' => 0,
+					],
+					'condition' => [
+						"_{$tab}_transform_skew_popover!" => '',
+					],
+					'selectors' => [
+						"{{WRAPPER}} > .elementor-widget-container{$hover_state}" => '--e-transform-skewX: {{SIZE}}deg;',
+					],
+					'frontend_available' => true,
+				]
+			);
+
+			$this->add_responsive_control(
+				"_{$tab}_transform_skewY_effect",
+				[
+					'label' => __( 'Skew Y', 'elementor' ),
+					'type' => Controls_Manager::SLIDER,
+					'range' => [
+						'px' => [
+							'min' => -360,
+							'max' => 360,
+						],
+					],
+					'default' => [
+						'size' => 0,
+					],
+					'condition' => [
+						"_{$tab}_transform_skew_popover!" => '',
+					],
+					'selectors' => [
+						"{{WRAPPER}} > .elementor-widget-container{$hover_state}" => '--e-transform-skewY: {{SIZE}}deg;',
+					],
+					'frontend_available' => true,
+				]
+			);
+
+			$this->end_popover();
+
+			$this->add_control(
+				"_{$tab}_transform_flipX_effect",
+				[
+					'label' => __( 'Flip Horizontal', 'elementor' ),
+					'type' => Controls_Manager::CHOOSE,
+					'options' => [
+						'transform' => [
+							'title' => __( 'Flip Horizontal', 'elementor' ),
+							'icon' => 'fa fa-align-left',
+						],
+					],
+					'prefix_class' => 'e-',
+					'selectors' => [
+						"{{WRAPPER}} > .elementor-widget-container{$hover_state}" => '--e-transform-flipX: -1',
+					],
+					'frontend_available' => true,
+				]
+			);
+
+			$this->add_control(
+				"_{$tab}_transform_flipY_effect",
+				[
+					'label' => __( 'Flip Vertical', 'elementor' ),
+					'type' => Controls_Manager::CHOOSE,
+					'options' => [
+						'transform' => [
+							'title' => __( 'Flip Vertical', 'elementor' ),
+							'icon' => 'fa fa-align-center',
+						],
+					],
+					'prefix_class' => 'e-',
+					'selectors' => [
+						"{{WRAPPER}} > .elementor-widget-container{$hover_state}" => '--e-transform-flipY: -1',
+					],
+					'frontend_available' => true,
+				]
+			);
+
+			if ( 'hover' === $tab ) {
+				$this->add_control(
+					'_hover_transform_transition',
+					[
+						'label' => __( 'Transition Duration (ms)', 'elementor' ),
+						'type' => Controls_Manager::SLIDER,
+						'range' => [
+							'px' => [
+								'min' => 100,
+								'max' => 20000,
```

Comment:
Are you sure that the max value should be 20,000? (20 seconds of transition time? - sounds a bot weird).

Reply:
After talking with Dor, I changed it to 10s.
done